libraries/botbuilder-core/src/memoryStorage.ts

```diff
@@ -66,7 +68,7 @@ export class MemoryStorage  implements Storage {
                     saveItem(key, newItem);
                 } else {
                     const oldItem: any = <any>JSON.parse(old);
-                    if (newItem.eTag === oldItem.eTag) {
+                    if (newItem.eTag === oldItem.eTag || !newItem.eTag) {
```

Comment:
>|| !newItem.eTag [](start = 54, length = 16)

I would move this to the outer if() since you're basically just defaulting newItem.eTag to '*'